src/dropdown/dropdown.js

```diff
@@ -54,7 +54,12 @@ angular.module('ui.bootstrap.dropdown', [])
   };
 
   $scope.$watch('isOpen', function( value ) {
-    self.$element.toggleClass( openClass, value );
+    if (value)  {
```

Comment:
$animate should have a promise, and the code after that should be inside the resolved function. Else the toggle function would be called before the dropdown is opened.